cmd/admin-handlers.go

```diff
@@ -1199,15 +1200,51 @@ func (a adminAPIHandlers) OBDInfoHandler(w http.ResponseWriter, r *http.Request)
 		finish()
 	}
 
+	deadlineSeconds := time.Duration(3600)
+	deadline := r.URL.Query().Get("deadlineseconds")
+	if deadline != "" {
+		deadlineSecondsInt, err := strconv.Atoi(deadline)
+		if err != nil {
+			errResp(err)
+			return
+		}
+		deadlineSeconds = time.Duration(deadlineSecondsInt)
+	}
+
+	deadlinedCtx, cancel := context.WithDeadline(ctx, time.Now().Add(deadlineSeconds*time.Second))
+
+	setCommonHeaders(w)
+	w.Header().Set(xhttp.ContentType, string(mimeJSON))
+	w.WriteHeader(http.StatusOK)
+
+	ticker := time.NewTicker(30 * time.Second)
+	defer ticker.Stop()
+
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				doPartialWrite()
+			case <-pulse:
+				doPartialWrite()
+			case <-obdDone:
+				break
+			case <-deadlinedCtx.Done():
+				break
+			}
+		}
+		doPartialWrite()
```

Comment:
travis build failure here as code is unreachable.

Reply:
this is fixed now